Documentation for 2.44.0 (#1334)
percona/pmm-doc: docs/release-notes/2.44.0.md

# Percona Monitoring and Management 2.44.0

> | **Release date** | December 90th, 2024                                                                                     |

Reviewer: something is wrong here :)

| -----------------| ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/index.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them, and execute database management operations regardless of whether your databases are located on-premises or in the cloud.

## Improved PostgreSQL 17 Metrics Collection

> PMM 2.44.0 strengthens monitoring capabilities for PostgreSQL 17, building on the recent support for Query Analytics introduced in PMM 2.443.2. 

Reviewer: missprint: 2.443.2

### [Tech Preview] Support for PSMDB  and Community MongoDB 8.0

The latest version of MongoDB, along with [Percona Server for MongoDB 8.0](https://www.percona.com/software/mongodb/percona-server-for-mongodb), brings [numerous improvements and significant performance enhancements](https://www.mongodb.com/docs/manual/release-notes/8.0/).

With this update, we are adding support for MongoDB 8 in PMM 2, including updates to `mongodb_exporter` to accommodate the revised metrics structure (e.g., `wiredTiger.concurrentTransactions` is now `queues.execution`).

In addition, the MongoDB **Oplog Details** dashboard is adapted to support MongoDB 8.0's new oplog metrics, with updated panels for **Oplog Buffered Operations** and **Buffer Capacity**.

![Updated Oplog Details dashboard](Oplog_Details.png)
This enhancement requires PMM Agent version 2.43.1 or later. Keep in mind that some dashboard metrics may need further updates to fully support MongoDB 8.0's new format.

## Fixed PBM metrics collection

We've resolved an issue where Percona Backup for MongoDB (PBM) metrics were not being scraped by default and required the `--enable-all-collectors` flag.

> With this fix, PBM metrics are now automatically collected when MongoDB services are added to PMM, without requiring additional configuration.

Reviewer 2: we also disabled pbm metrics collection for PMM Clients 2.43.0 and 2.43.1 as there were a memory leak. So now pbm metrics collection is supported only for PMM Clients starting 2.43.2

# Percona Monitoring and Management 2.44.0

| **Release date** | December 9th, 2024                                                                                     |
| -----------------| ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/index.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them, and execute database management operations regardless of whether your databases are located on-premises or in the cloud.

## Support for version 8.0 of Percona Server for MongoDB (PSMDB), MongoDB Community, and MongoDB Enterprise

> The latest version of MongoDB, along with the upcoming [PSMDB 8.0](https://www.percona.com/software/mongodb/percona-server-for-mongodb), introduce [numerous improvements and significant performance enhancements](https://www.mongodb.com/docs/manual/release-notes/8.0/).

Reviewer 3: ```suggestion
The latest version of MongoDB community edition, along with the upcoming [PSMDB 8.0](https://www.percona.com/software/mongodb/percona-server-for-mongodb), introduce [numerous improvements and significant performance enhancements](https://www.mongodb.com/docs/manual/release-notes/8.0/).
```

> We have updated PMM 2 to support these new versions, including changes to `mongodb_exporter` to accommodate the revised metrics structure (e.g., `wiredTiger.concurrentTransactions` is now `queues.execution`).

Reviewer 3: ```suggestion
We have updated PMM 2 to include support for these new versions, including changes to `mongodb_exporter` to accommodate the revised metrics structure (e.g., `wiredTiger.concurrentTransactions` is now `queues.execution`).
```

> The MongoDB **Oplog Details** dashboard is also adapted to support MongoDB 8.0's new oplog metrics, with updated panels for **Oplog Buffered Operations** and **Buffer Capacity**.

Reviewer 3: ```suggestion
The MongoDB **Oplog Details** dashboard has also been adapted to support MongoDB 8.0's new oplog metrics, with updated panels for **Oplog Buffered Operations** and **Buffer Capacity**.
```

![Updated Oplog Details dashboard](../_images/Oplog_Details.png)

!!! hint alert alert-success "Important notes"
    - This enhancement requires PMM Agent version 2.43.1 or later.

    - MongoDB 8.0 introduces significant changes to its internal metrics structure. While we have updated PMM’s built-in dashboards to reflect these changes, you may need to update any custom dashboards to align with the new metrics.

## Improved PostgreSQL 17 metrics collection

> PMM 2.44.0 strengthens monitoring capabilities for PostgreSQL 17, building on the recent support for Query Analytics introduced in PMM 2.43.2.

Reviewer 3: can we link to those updates or maybe remind the user of a few?

# Percona Monitoring and Management 2.44.0

| **Release date** | December 9th, 2024                                                                                     |
| -----------------| ----------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/index.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

It enables you to observe the health of your database systems, explore new patterns in their behavior, troubleshoot them, and execute database management operations regardless of whether your databases are located on-premises or in the cloud.

> ## Support for Percona Server for MongoDB 8.0  and Community MongoDB 8.0

Reviewer 4: "Community MongoDB 8.0" 🤔  is this how we mention Mongo? Because, in theory, this, this will work with any Mongo (community or Enterprise). Maybe it's worth mentioning that it's explicitly to highlight that you can use PMM with the $(commercial) version of MongoDB too.

Reviewer 5: We use these:
- MongoDB Community 
- MongoDB Enterprise Advanced 

In that case could be:
## Support for Percona Server for MongoDB, MongoDB Community, and MongoDB Enterprise 8.0

In addition, the MongoDB **Oplog Details** dashboard is adapted to support MongoDB 8.0's new oplog metrics, with updated panels for **Oplog Buffered Operations** and **Buffer Capacity**.

![Updated Oplog Details dashboard](Oplog_Details.png)

!!! hint alert alert-success "Important notes"
    - This enhancement requires PMM Agent version 2.43.1 or later.

    - MongoDB 8.0 introduces significant changes to its internal metrics structure. While PMM’s built-in dashboards have been updated to reflect these changes, you may need to update any custom dashboards to align with the new metrics.
 
## Improved PostgreSQL 17 Metrics Collection

PMM 2.44.0 strengthens monitoring capabilities for PostgreSQL 17, building on the recent support for Query Analytics introduced in PMM 2.43.2.
With this update, we've revised queries tailored to PostgreSQL 17's updated schema to ensure that metrics are accurately collected and displayed in the **PostgreSQL Overview** dashboard.

Columns like `checkpoints_timed` in the `stat_bgwriter` collector are now also correctly recognized.

> ## Fixed PBM metrics collection

Reviewer 4: ```suggestion
## Fixed metrics collection for MongoDB backups 
```
? To avoid using abbreviations in the title.

    - MongoDB 8.0 introduces significant changes to its internal metrics structure. While we have updated PMM’s built-in dashboards to reflect these changes, you may need to update any custom dashboards to align with the new metrics.

>     - When using `--enable-all-collectors`, monitor memory usage carefully with MongoDB 8.0+ clusters, especially in sharded environments with multiple collections. Consider using specific collectors for better performance if your environment frequently creates new collections.

Reviewer: not with Mongo 8 only, but with Mongo 7, so i'd remove v8
